## Profitable App Profiles for the App Store and Google Play Markets.

The project is aimed to analyze applications from App Store and Google play markets and find out the most profitable to help the software-developing company make a data-driven decision.

Due to the fact, that there are only free applications considered by developers, the profitability will be based on the user involvement and popularity as our source of revenue is in-app advertisement. Therefore, our aim, as data analysts, is to help to uderstand what type of applications are popular in App Store and Google Market.


In [1]:
#AppleStore dataset

from csv import reader

opened_file = open('AppleStore.csv',encoding='utf8')
read_file = reader(opened_file)
ios = list(read_file)
ios_header = ios[0]
ios = ios[1:]

#GooglePlayMarket dataset

opened_file = open('googleplaystore.csv')
read_file = reader(opened_file)
android = list(read_file)
android_header = android[0]
android = android[1:]

In [2]:
def explore_data(dataset, start, end, rows_and_columns = False):
    dataset_slice = dataset[start:end]
    for row in dataset_slice:
        print(row)
        print('\n')
    
    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))
print(ios_header)
print('\n')
explore_data(ios, 0, 5, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']


Number of rows: 7197
Number of columns: 16


In [3]:
print(android_header)
print('\n')
explore_data(android, 1,3,True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


In [4]:
print(android[10472])

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


In [5]:
del android[10472]

In [6]:
print(android[10472])

['osmino Wi-Fi: free WiFi', 'TOOLS', '4.2', '134203', '4.1M', '10,000,000+', 'Free', '0', 'Everyone', 'Tools', 'August 7, 2018', '6.06.14', '4.4 and up']


In [7]:
for app in android:
    name = app[0]
    if name == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


In [8]:
duplicate_apps = []
unique_apps = []

for app in android:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)
        
print('Number of duplicate apps:', len(duplicate_apps))
print('\n')
print('Examples of duplicate apps:', duplicate_apps[:15])
        

Number of duplicate apps: 1181


Examples of duplicate apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


In [9]:
reviews_max ={}

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    if name in reviews_max and (reviews_max[name] < n_reviews):
        reviews_max[name] = n_reviews
    elif name not in reviews_max:
        reviews_max[name] = n_reviews
    

In [10]:
print('Expected length:', len(android) -1181 )
print('Actual length:', len(reviews_max))


Expected length: 9659
Actual length: 9659


In [11]:
android_clean  = []
already_added = []

for app in android:
    name = app[0]
    n_reviews = float(app[3]) 
    if (n_reviews == reviews_max[name]) and ( name not in already_added):
        android_clean.append(app)
        already_added.append(name)
        
    

In [12]:
explore_data(android_clean, 1,3, True)

['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9659
Number of columns: 13


In [13]:
def is_english(string):
    for character in string:
        if ord(character) > 127:
            return False
    return True

print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(is_english('Instagram'))
print(is_english(':)'))
print(is_english('Instachat 😜'))
print(is_english('Docs To Go™ Free Office Suite'))



False
True
True
False
False


In [14]:
def is_english(string):
    non_ascii = 0
    
    for character in string:
        if ord(character) > 127:
            non_ascii += 1
            
    if non_ascii > 3:
            return False
    else:
            return True
        
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热'))
        
        

True
True
False


In [15]:
android_english = []
ios_english = []

for app in android_clean:
    name = app[0]
    if is_english(name):
        android_english.append(app)
        
for app in ios:
    name = app[1]
    if is_english(name):
        ios_english.append(app)
        
explore_data(android_english, 1, 3, True)
print('\n')
explore_data(ios_english, 1, 3, True)


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 6183
Number of columns: 16


In [16]:
android_final = []
ios_final = []

for app in android_english:
    price = app[7]
    if price == '0':
        android_final.append(app)
        
for app in ios_english:
    price = app[4]
    if price =='0.0':
        ios_final.append(app)
        
print(len(android_final))
print(len(ios_final))

        


8864
3222


After cleaning the data, we are on the next phase to analyze what kind of apps are most popular based on the fact that our revenue will be generated by the in-app advertisements. Therefore, it is crucial to to attract many users in Google Play and App Store.
Initial step is to build a minimal Android verison of the app for Google Play. After analyzing the profitability of the application to continue develping it, and, eventually, introducing it to the App Store.

After inspecting both data sets, we see that we will compare genres, categiry and app ratings to find out what is most popular among users.

In [17]:
def freq_table(dataset,index):
    table = {}
    total = 0
    # print('dataset: ', dataset)
    # print('table start: ', table)
    # print('\n')
    for row in dataset:
        # print('row: ', row)
        # print('\n')
        total += 1
        value = row[index]
        # print('value :', value)
        # print('\n')
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
            
    # print('table end: ', table)
    # print('\n')
    
    table_percentages = {}
    for key in table:
        percentage = (table[key]/total)*100
        table_percentages[key] = percentage
    
    return table_percentages


def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        # print('table_key:', table[key])
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
        # print(table_display)
        
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])
    


            

In [18]:
display_table(ios_final, -5)
# display_table([data for data in ios_final[:3]], -5)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


Taken the analysis of the most commom genre in App Store among free English apps, the majority percentage is taken by Games with 58.16%. The runner-up is the Entertainment with only 7.88%. The drastical difference between the first and second most common apps in Apple Store make the undoubtful dominance of the Gaming genre in our dataset and apps used for the entertainemnt purposes in general taking into consideration that the third place in the most common genre is taken by Photo & Video.
However, even within the Game genre itself there could be different types/categories of games, such as educational, for kids, logical, etc.
Moreover, solely based on the frequency showed, this app profile cannot be recommended, as we are not certain that gaming applications have larger number of users.

Below, we have genertaed the frequency table for Category and Genres column of the Google data set. The analysis is limited to consideration of only english free applications.
Based on the created frequesncty table the most common Genre in Google Play are Tools, entertainment, and Education, which present a significantly different trend compared to the App Store with a slight percentage difference between them.
The most common category is Family with 18.90%. The runner-up is Game cateogry taking 9.72% among all free english Google Play applications.
The generated frequency tables, however, do not reveal what app genres have most users, which is a crucial part of our decision-making process.

In [19]:
display_table(android_final, -4)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

In [20]:
display_table(android_final, 1)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

In [21]:
genres_ios = freq_table(ios_final, -5)

    
for genre in genres_ios:
        total = 0     # sum of user ratings (#of ratings) specific to each genre
        len_genre = 0 # number of apps specific to each genre
    
        for app in ios_final:
            genre_app = app[-5]
            if genre_app == genre:
                n_ratings = float(app[5])
                total += n_ratings
                len_genre += 1
                
        avg_n_rating = total/len_genre
        print(genre, ':', avg_n_rating)
        
      



                 
        
        

Health & Fitness : 23298.015384615384
Photo & Video : 28441.54375
Travel : 28243.8
Music : 57326.530303030304
Medical : 612.0
Weather : 52279.892857142855
Games : 22788.6696905016
Navigation : 86090.33333333333
Social Networking : 71548.34905660378
Utilities : 18684.456790123455
Food & Drink : 33333.92307692308
Education : 7003.983050847458
Lifestyle : 16485.764705882353
Productivity : 21028.410714285714
Sports : 23008.898550724636
Reference : 74942.11111111111
Finance : 31467.944444444445
Book : 39758.5
Catalogs : 4004.0
News : 21248.023255813954
Entertainment : 14029.830708661417
Shopping : 26919.690476190477
Business : 7491.117647058823


Based on the calculations of the average number of user ratings for each genre, we can see that Navigation apps are most popular among users in App Store (considering free English applications), followed by Social Networks and Reference.

In [22]:
for app in ios_final:
    if app[-5] == 'Navigation':
        print(app[1],':', app[5]) # print name and number of rating
        

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


However, having a closer look at the breakdown of the applications, we can clearly see that the dominance of Navigation genre comes solely from Waze and Google Maps, which alltogether cover alomost half a million of ratings. Therefore, picking Navigation as a genre to invest in is not accurate.
Follwoing the same routine we can get breakdowns of the following two popular genres: Music and Weather:

In [23]:
for app in ios_final:
    if app[-5] == 'Music':
        print(app[1], ':', app[5])

Pandora - Music & Radio : 1126879
Spotify Music : 878563
Shazam - Discover music, artists, videos & lyrics : 402925
iHeartRadio – Free Music & Radio Stations : 293228
SoundCloud - Music & Audio : 135744
Magic Piano by Smule : 131695
Smule Sing! : 119316
TuneIn Radio - MLB NBA Audiobooks Podcasts Music : 110420
Amazon Music : 106235
SoundHound Song Search & Music Player : 82602
Sonos Controller : 48905
Bandsintown Concerts : 30845
Karaoke - Sing Karaoke, Unlimited Songs! : 28606
My Mixtapez Music : 26286
Sing Karaoke Songs Unlimited with StarMaker : 26227
Ringtones for iPhone & Ringtone Maker : 25403
Musi - Unlimited Music For YouTube : 25193
AutoRap by Smule : 18202
Spinrilla - Mixtapes For Free : 15053
Napster - Top Music & Radio : 14268
edjing Mix:DJ turntable to remix and scratch music : 13580
Free Music - MP3 Streamer & Playlist Manager Pro : 13443
Free Piano app by Yokee : 13016
Google Play Music : 10118
Certified Mixtapes - Hip Hop Albums & Mixtapes : 9975
TIDAL : 7398
YouTube Mu

In [24]:
for app in ios_final:
    if app[-5] == 'Weather':
        print(app[1], ':', app[2])

The Weather Channel: Forecast, Radar & Alerts : 199734272
The Weather Channel App for iPad – best local forecast, radar map, and storm tracking : 59101184
WeatherBug - Local Weather, Radar, Maps, Alerts : 100524032
MyRadar NOAA Weather Radar Forecast : 127284224
AccuWeather - Weather for Life : 181941248
Yahoo Weather : 123047936
Weather Underground: Custom Forecast & Local Radar : 99829760
NOAA Weather Radar - Weather Forecast & HD Radar : 108172288
Weather Live Free - Weather Forecast & Alerts : 101524480
Storm Radar : 93304832
QuakeFeed Earthquake Map, Alerts, and News : 70262784
Moji Weather - Free Weather Forecast : 86968320
Hurricane by American Red Cross : 54710272
Forecast Bar : 129855488
Hurricane Tracker WESH 2 Orlando, Central Florida : 29992960
FEMA : 8147968
iWeather - World weather forecast : 30547968
Weather - Radar - Storm with Morecast App : 185079808
Yurekuru Call : 56609792
Weather & Radar : 94449664
WRAL Weather Alert : 68850688
Météo-France : 124090368
JaxReady : 8

Based on the Music breakdown there could be seen the similar pattern, where a few big players like Pandora, Spotify and Shazam are significantly influencing the average of user ratings.
On the other hand, weather applications sound quite promising, but, users will not spend a lot of time in the application just to check the weather forecasts. Therefore, it does not fit the initial aim to make profit from in-app adds. Moreover, retrieving accurate and live information will require using more sofisticated and probably non-free services.

In [25]:
display_table(android_final, 5)

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


In [26]:
category_android = freq_table(android_final, 1)

for category in category_android:
    total = 0 # sum of installs specific to each genre
    len_category = 0 #number of apps specific to each genre
    
    for app in android_final:
        category_app = app[1]
        if category_app == category:
            n_installs = app[5]
            n_installs = n_installs.replace('+','')
            n_installs = n_installs.replace(',', '')
            n_installs = float(n_installs)
            total += n_installs
            len_category += 1

    avg_n_installs = total/len_category
    print(category, ':', avg_n_installs)

HEALTH_AND_FITNESS : 4188821.9853479853
BUSINESS : 1712290.1474201474
HOUSE_AND_HOME : 1331540.5616438356
DATING : 854028.8303030303
SOCIAL : 23253652.127118643
MEDICAL : 120550.61980830671
COMICS : 817657.2727272727
ENTERTAINMENT : 11640705.88235294
SHOPPING : 7036877.311557789
WEATHER : 5074486.197183099
ART_AND_DESIGN : 1986335.0877192982
MAPS_AND_NAVIGATION : 4056941.7741935486
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PARENTING : 542603.6206896552
PRODUCTIVITY : 16787331.344927534
VIDEO_PLAYERS : 24727872.452830188
BEAUTY : 513151.88679245283
GAME : 15588015.603248259
BOOKS_AND_REFERENCE : 8767811.894736841
FAMILY : 3695641.8198090694
AUTO_AND_VEHICLES : 647317.8170731707
PHOTOGRAPHY : 17840110.40229885
LIBRARIES_AND_DEMO : 638503.734939759
PERSONALIZATION : 5201482.6122448975
FINANCE : 1387692.475609756
COMMUNICATION : 38456119.167247385
EDUCATION : 1833495.145631068
SPORTS : 3638640.1428571427
FOOD_AND_DRINK : 1924897.7363636363
NEWS_AND_MAGAZINES : 954917